### PR TITLE
gojirun ui header navigation and authentication set

### PR DESCRIPTION
**Pull request by SunkaraboinaPraveenKumar**
###Use .env.example and convert to .env.local file while running to enable authentication.

> I have Integrated Clerk Authentication before entering to home page

> The header section is updated with highscores, settings, and help functionalities by adding seperate routes to them.

> navigation/header is added in layout.jsx instead of page.jsx for convinience of navigation through all routes of app.

> Implemented Settings section with use name and option to edit theme there also via a toggle button.

> Implemented Help section with FAQs as drop down

> Highscores section to get to know user's best score as of know using local storage.

> Made it Mobile responsive

### CODE REVIEW

File: gojirun/app/layout.jsx
Comment (allanoguis) [inline gojirun/app/layout.jsx:1]: Good take! Very nice

File: gojirun/components/engine.jsx
Comment (allanoguis) [inline gojirun/components/engine.jsx:1]: This one is nice. I learned from this

Comment (SunkaraboinaPraveenKumar) [reply gojirun/components/engine.jsx:1]: Thank You

**Review by allanoguis — CHANGES_REQUESTED** [review]
Hi @SunkaraboinaPraveenKumar may I ask for a revision or update on the files so we can make sure that this one is running?

**Comment by SunkaraboinaPraveenKumar** [comment]
Its working well on my side. Ok do one thing goto clerk.com and create a new project and use your own api keys you can keep NEXT_PUBLIC_CLERK_SIGN_IN_URL=/sign-in
NEXT_PUBLIC_CLERK_SIGN_UP_URL=/sign-up
Theses two as it is and change other two which are above these two in .env.local that you have changed from .env.example

**Comment by allanoguis** [comment]
Got it. I'll do that now

**Review by allanoguis — APPROVED** [review]

**Review by vaibhav-gosavi — APPROVED** [review]

**Comment by SunkaraboinaPraveenKumar** [comment]
Bro can I further conribute for same project as can get another hacktober PR done or i have to choose different one all together

**Comment by Parth-G27** [comment]
@SunkaraboinaPraveenKumar Join the discord server for better understanding.. 
https://discord.gg/4eYDmch

**Comment by Parth-G27** [comment]
@vaibhav-gosavi @vprathap21 It would be great if you guys can also join the discord server. 
Will help in better co ordination

**Comment by allanoguis** [comment]
you can try this link: https://discord.gg/UrQGsZvE

**Review by Parth-G27 — COMMENTED** [review]
Clerk Authentification is very helpful (sign-in & sign-up). We will use clerk part in the application,  once the application is full-stack, right now it will just cause more conflicts in the code as we need to setup the API call for that as well when the user sign-in & when user sign-out. 
amazing work @SunkaraboinaPraveenKumar . 
also the /help & /highscore page addition is a good idea.

**Comment by SunkaraboinaPraveenKumar** [comment]
Thank you Brother

**Pull request closed without merging**